drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -679,6 +700,82 @@ def has_submission_deadline_passed(self):
         else:
             return False
 
+    @property
+    def closed(self):
+        """
+        Is the student still allowed to submit answers?
+        """
+        if self.used_all_attempts:
+            return True
+        if self.has_submission_deadline_passed:
+            return True
+
+        return False
+
+    @property
+    def is_attempted(self):
+        """
+        Has the problem been attempted?
+        """
+        return self.attempts > 0
+
+    @property
+    def is_finished(self):
+        """
+        Returns True if answer is closed or answer is correct.
+        """
+        return self.closed or self.is_correct
+
+    @property
+    def used_all_attempts(self):
+        """
+        Returns True if student has used all allowed attempts.
+        """
+        if self.max_attempts is None or self.max_attempts == 0:
+            return False
+        return self.attempts >= self.max_attempts
+
+    @property
+    def is_correct(self):
+        """
+        Returns True if answer is correct.
+        """
+        return self._is_answer_correct()
+
+    @property
+    def is_answer_available(self):
+        """
+        Is student allowed to see an answer?
+        """
+        if self.mode != Constants.ASSESSMENT_MODE:
+            return False
+        _answer_available = False
+        if self.showanswer == SHOWANSWER.NEVER:
+            _answer_available = False
+        elif self.showanswer == SHOWANSWER.ATTEMPTED:
```

Comment:
because the service does not exist I got ```NoSuchServiceError```

```python
raise NoSuchServiceError(f"Service {service_name!r} was not requested.")
xblock.exceptions.NoSuchServiceError: Service 'user' was not requested.
```
can you mention if there is another way to call ```user``` service?

Thanks

Reply:
I refactored the code to solve  the linter error and ```staff``` permissions check